src/Uno.SourceGeneration.Intellisense/SourceGenerationIntellisenseFixerPackage.cs

```diff
@@ -65,27 +69,57 @@ private void BuildEventsOnOnBuildDone(vsBuildScope scope, vsBuildAction action)
 					return; // not interesting
 			}
 
-			var documents = Interlocked.Exchange(ref _changedDocuments, new ConcurrentBag<Document>());
+			foreach(var project in GetProjectsUsingCodeGen())
+			{
+				try
+				{
+					var folderItem = project.ProjectItems.AddFolder("__temp-to-delete");
+					folderItem.Remove();
+				}
+				catch(Exception ex)
+				{
+					Debug.WriteLine($"Error: {ex.Message}");
+				}
+			}
+		}
 
-			// Fixing only one document per project is enough
-			var oneDocumentPerProject = documents
-				.GroupBy(d => d.ProjectItem.ContainingProject)
-				.Select(p => p.FirstOrDefault(d => File.Exists(d.FullName))) // we want non-deleted documents
-				.Where(d => d != null); // prevent NRE
+		private IEnumerable<Project> GetProjectsUsingCodeGen(Project[] projects = null)
+		{
+			projects = projects ?? _dte.Solution.Projects.OfType<Project>().ToArray();
 
-			foreach (var document in oneDocumentPerProject)
+			foreach (var project in projects)
 			{
-				var property= document.ProjectItem.Properties.Item("ItemType");
-				var previousValue = property.Value as string;
-				if (previousValue == "Compile")
+				_solution.GetProjectOfUniqueName(project.FullName, out var hierarchy);
+
+				if (hierarchy is IVsBuildPropertyStorage storage)
 				{
-					property.Value = "None";
-					property.Value = previousValue;
+					storage.GetPropertyValue("UnoSourceGeneration_FixIntellisense", string.Empty, (uint)_PersistStorageType.PST_PROJECT_FILE, out var value);
+
+					if (!string.IsNullOrWhiteSpace(value) &&
+						bool.TryParse(value, out var fixIntellisense) &&
+						fixIntellisense)
+					{
+						yield return project;
+					}
 				}
 
-				// Force save project to avoid annoying the user about saving it
-				document.ProjectItem.ContainingProject.Save();
+				var subProjects = project.ProjectItems
+					.OfType<ProjectItem>()
+					.Where(p=> p.Kind.Equals("{66A26722-8FB5-11D2-AA7E-00C04F688DDE}", StringComparison.OrdinalIgnoreCase))
```

Comment:
Where is this Guid coming from? Can you document it ?

Reply:
Yep I'll do.

Reply:
Done :-)